### PR TITLE
mgr/dashboard: delete-ceph-authx

### PR DESCRIPTION
**Pull request by Pegonzal**
Add option to delete users on ceph authx

image

Fixes: https://tracker.ceph.com/issues/59365
Signed-off-by: Pedro Gonzalez Gomez [pegonzal@redhat.com](mailto:pegonzal@redhat.com)

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/controllers/_crud.py [hunk 1]
```diff
@@ -64,7 +64,8 @@ class TableAction(NamedTuple):
     name: str
     permission: str
     icon: str
-    routerLink: str  # redirect to...
+    routerLink: str = ''  # redirect to...
+    click: str = ''
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/controllers/_crud.py:68]: Why an extra `click` variable? Correct me if I am wrong but if this is for assuming the action that is going to perform (create, edit, delete etc), then the `name` itself should be enough right?

**Review by nizamial09 — COMMENTED** [review]
Looks good to me @Pegonzal 

Although the delete shouldn't happen that quickly. When you click delete it should open up a simple modal and ask the user to confirm the deletion. Otherwise its a surprise for him. Because I was expecting a modal but got suprised that it got deleted immediately. But relieved that it actually got deleted xD.

File: src/pybind/mgr/dashboard/controllers/_crud.py (hunk 1 above)
Comment (Pegonzal) [reply src/pybind/mgr/dashboard/controllers/_crud.py:68]: The idea is to have a variable, `click`, that you can set the name to match your frontend function, and then to map this click variable into the `CdTableAction` `click` variable that is generated from the `meta`. 

I thought this was a good approach to have functions that don't use a form, and they just trigger a function, let me know what you think Nizam

**Comment by Pegonzal** [comment]
> Looks good to me @Pegonzal
> 
> Although the delete shouldn't happen that quickly. When you click delete it should open up a simple modal and ask the user to confirm the deletion. Otherwise its a surprise for him. Because I was expecting a modal but got suprised that it got deleted immediately. But relieved that it actually got deleted xD.

Wops, you are right. I added the modal to ask for confirmation before the deletion, thanks for the review @nizamial09

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/crud-table/crud-table.component.ts [hunk 2]
```diff
@@ -86,6 +98,30 @@ export class CRUDTableComponent implements OnInit {
...
+        this.meta.actions[i].click = this[this.meta.actions[i].click.toString()].bind(this);
+      }
+    }
+  }
+
+  delete() {
+    const selectedKey = this.selection.first()[this.meta.columnKey];
+    this.modalRef = this.modalService.show(CriticalConfirmationModalComponent, {
+      itemDescription: $localize `${this.meta.columnKey}`,
+      itemNames: [`${selectedKey}`],
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/crud-table/crud-table.component.ts:112]: ?
```suggestion
      itemNames: [selectedKey],
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/crud-table/crud-table.component.ts [hunk 3]
```diff
@@ -86,6 +98,30 @@ export class CRUDTableComponent implements OnInit {
...
+      if (this.meta.actions[i].click.toString() !== '') {
+        this.meta.actions[i].click = this[this.meta.actions[i].click.toString()].bind(this);
+      }
+    }
+  }
+
+  delete() {
+    const selectedKey = this.selection.first()[this.meta.columnKey];
+    this.modalRef = this.modalService.show(CriticalConfirmationModalComponent, {
+      itemDescription: $localize `${this.meta.columnKey}`,
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/crud-table/crud-table.component.ts:111]:
```suggestion
      itemDescription: $localize`${this.meta.columnKey}`,
```

**Review by nizamial09 — APPROVED** [review]
LGTM! Thank you @Pegonzal, some minor nits.

**Review by pereman2 — APPROVED** [review]
lgtm

**Pull request merged**